plutus-core/plutus-ir/Language/PlutusIR/TypeCheck/Internal.hs

```diff
@@ -335,33 +390,32 @@ withVarsOfBinding r (DatatypeBind _ dt) k = do
     -- generate all the definitions
     (_tyconstrDef, constrDefs, destrDef) <- compileDatatypeDefs r (original dt)
     -- ignore the generated rhs terms of constructors/destructor
-    let structorDecls = fmap (PIR.defVar) $ destrDef:constrDefs
+    let structorDecls = PIR.defVar <$> destrDef:constrDefs
     -- normalize, then rename, then only introduce the vardecl to scope
     foldr normRenameScope k structorDecls
     where
       normRenameScope :: VarDecl TyName Name uni (Provenance a)
-                      -> TypeCheckM uni e res -> TypeCheckM uni e res
+                      -> TypeCheckM uni o e res -> TypeCheckM uni o e res
       normRenameScope v acc = do
           normRenamedTy <- rename =<< (normalizeTypeM $ varDeclType v)
           withVar (varDeclName v) (void <$> normRenamedTy) acc
 
 
 withVarsOfBindings :: Foldable t => Recursivity -> t (Binding TyName Name uni a)
-                   -> TypeCheckM uni e res -> TypeCheckM uni e res
-withVarsOfBindings r bs k = foldr (withVarsOfBinding r) k bs
-
+                   -> TypeCheckM uni o e res -> TypeCheckM uni o e res
+withVarsOfBindings = flip . foldr . withVarsOfBinding
```

Comment:
I think this is quite a lot less readable than the old version. Being point-free isn't necessarily good, and especially stuff like `flip` is pretty hard to keep in your head right.

Reply:
If I was trying to edit this function the first thing I would do is put the variables back!